### PR TITLE
add torch.no_grad() for block swap

### PR DESCRIPTION
**Pull request by kohya-ss**
This may solve a potential issue and prevent loss=NaN in long training. Further testing is needed to see if existing training works as well as before.

### CODE REVIEW

**Comment by FurkanGozukara** [comment]
> This fix appears to enable multi-GPU training of DDP, but further testing is required.

with flux we were never able to do multi gpu training with block swap

so it is possible?

**Comment by FurkanGozukara** [comment]
> > so it is possible?
> 
> I think so. I will add a new branch to support this on sd-scripts repo today.

awesome

**Comment by xzuyn** [comment]
I've been using this on a single amd gpu on ubuntu for a bit, training qwen image. Haven't noticed any problems.

File: modules/custom_offloading_utils.py [hunk 1]
```diff
@@ -56,19 +56,20 @@ def swap_weight_devices_cuda(device: torch.device, layer_to_cpu: nn.Module, laye
...
-            cuda_data_view.record_stream(stream)
-            module_to_cpu.weight.data = cuda_data_view.data.to("cpu", non_blocking=True)
-
-        stream.synchronize()
-
-        # cpu to cuda
-        for module_to_cpu, module_to_cuda, cuda_data_view, cpu_data_view in weight_swap_jobs:
-            cuda_data_view.copy_(module_to_cuda.weight.data, non_blocking=True)
-            module_to_cuda.weight.data = cuda_data_view
+    with torch.no_grad():
```
Comment (hameerabbasi) [inline modules/custom_offloading_utils.py:59]:
```suggestion
    with torch.inference_mode():
```

See https://discuss.pytorch.org/t/pytorch-torch-no-grad-vs-torch-inference-mode/134099/2
